### DISCUSSION TITLE
llvm_mode not detected

### DISCUSSION THREAD

**Discussion by nsri7** [AFLplusplus/AFLplusplus#1414] [General]
1) Followed the instructions from `docs/INSTALL.md`  and installed `llvm-12` accordingly.
` $ sudo apt-get install -y lld-12 llvm-12 llvm-12-dev clang-12`

2) When tried to build, got the below message, and thus, afl-clang-lto was not built. 
```
make distrib
...
GNUmakefile.llvm:60: [!] llvm_mode needs llvm-config, which was not found. Set LLVM_CONFIG to its path and retry.
[!] llvm_mode detected an old version of llvm, upgrade to at least 9 or preferable 11!
[+] llvm_mode detected llvm < 11, afl-lto LTO will not be build.
...
```
3) But, `llvm-*-12` binaries were present in `/usr/bin/`  after installation 

4) However,

- `GNUmakefile.llvm:42` `LLVM_CONFIG ?= llvm-config` for non OpenBSD systems.
- `GNUmakefile.llvm:52` `LLVM_HAVE_LTO = $(shell $(LLVM_CONFIG) --version 2>/dev/null | egrep -q '^1[1-9]' && echo 1 || echo 0 )`  - this shall result 0 because `llvm-config` is not present in the prefix paths mentioned in the script. 
```
$ llvm-config-12 --version
12.0.0

$ llvm-config --version
Command 'llvm-config' not found, but can be installed with:

$llvm-config --version 2>/dev/null | egrep -q '^1[1-9]' && echo 1 || echo 0
0
```
5) Hence, we're hitting the below condition at `GNUmakefile.llvm:96` although `llvm-config-12` and `llvm-lto-12` are present in `/usr/bin/`
```
ifeq "$(LLVM_LTO)" "0"
 $(info [+] llvm_mode detected llvm < 11, afl-lto LTO will not be build.)`
endif

```
Any quick work around for this, please ?

**Comment by vanhauser-thc** [comment]
yes and that is documented. set LLVM_CONFIG to llvm-config-12

**Reply by nsri7** [reply]
`LLVM_CONFIG=llvm-config-12 make distrib`

```
[+] llvm_mode detected llvm 11+, enabling afl-lto LTO implementation
[+] shmat seems to be working.
[*] Checking for working 'llvm-config'...
[*] Checking for working '/usr/lib/llvm-12/bin/clang'...
[*] Checking for matching versions of '/usr/lib/llvm-12/bin/clang' and 'llvm-config-12'
[*] We have llvm-config version 12.0.0 with a clang version 12.0.0, good.

```
